api/catalog/api/views/media_views.py

```diff
@@ -143,24 +157,69 @@ def _get_user_ip(request):
         return ip
 
     @staticmethod
-    def _get_proxied_image(image_url, width=settings.THUMBNAIL_WIDTH_PX):
-        if width is None:  # full size
-            proxy_upstream = f"{settings.THUMBNAIL_PROXY_URL}/{image_url}"
-        else:
-            proxy_upstream = (
-                f"{settings.THUMBNAIL_PROXY_URL}/"
-                f"{settings.THUMBNAIL_WIDTH_PX},fit/"
-                f"{image_url}"
-            )
+    def _thumbnail_proxy_comm(
+        path: str,
+        params: dict,
+        headers: List[
+            tuple[str, str]
+        ] = None,  # ``List`` because there is a ``list`` function
```

Comment:
Setting it to `None` by default would require wrapping this in `Optional` for the type to be accurate, no?

Could just default to an empty list, alternatively. The existing code will work exactly as is, although the `if headers` is a little redundant considering iteration over an empty list is a no-op anyway and doesn't need to be guarded anymore if we don't use the `None` default.

Reply:
Linters generally complain if we use a mutable such as [] as the default value of an argument (for good reason). It might just be better to make the type Optional.

Reply:
Ahh, of course! I'd seen that linting error before but never ingrained why it was bad, it turns out Python re-uses the default array instance between invocations:

```
>>> def arr_default(a = []):
...     a.append(1)
...     return a
... 
>>> a = arr_default()
>>> a
[1]
>>> arr_default()
[1, 1]
>>> a
[1, 1]
```

Reply:
As long as you don't mutate the mutable inside the function, there's technically no issue with doing this. But considering that it's a potential pit that's easy to fall into, best to avoid it altogether and use `None`.

Reply:
It'd be nice if Python had a sort of "readonly list" for times like these... Oh wait! A tuple? :grin:

Reply:
Tuples aren't readonly.

Reply:
Yeah, I was wrong. So I immediately deleted it 😆 . It's certainly feasible to use it instead of an array as the default argument. It would certainly be anti-conventional but there are not _technical_ problems with doing so.

Reply:
Updated code to use tuple in ff1c7df.

Reply:
Ah whoops, I didn't see the comment disappear, sorry!

I wonder if it would even be anti-conventional. If you have access to GitHub codesearch, there's a lot of results of it being used as the default for an iterable argument: https://cs.github.com/?scopeName=All+repos&scope=&q=language%3Apython+%22%3D+tuple%28%29%2C%22

Reply:
FWIW that link does not work for me 😅 seems I have to sign up for the service this corresponds to. If we wanted to make it generic for both lists and tuples we could use a [`typing.Iterable`](https://docs.python.org/3/library/typing.html#typing.Iterable) type instead. But having it explicitly as a tuple is also good 🙂